examples/adjoints/adjoints_ex5/heatsystem.h

```diff
@@ -140,6 +138,9 @@ class HeatSystem : public FEMSystem
   std::vector<Number> R_plus_dp;
   std::vector<Number> R_minus_dp;
 
+  // A vector to hold the possibly adaptive deltats
+  std::vector<Number> deltat_vector;
```

Comment:
Hmm... this works for now, but in the future we definitely want to have this accessed via some SolutionHistory APIs.  Note that we'd have to change this user code for any adaptive case where we're doing multiple sweeps through time, for example.

Reply:
Yeah, I mentioned in the other PR that we might want to do our adjoint sensitivities in a separate class like we do for ARef and ARes. Elements of the calculation like these can be members of that class, that interact with the timesolver and mutate accordingly.